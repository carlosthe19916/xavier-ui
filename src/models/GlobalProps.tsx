export interface GlobalProps {
    history: any;
    location: any;
    match: any;
}
