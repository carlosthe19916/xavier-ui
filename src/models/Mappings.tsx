export interface FlagAssessmentModel {
    flag: string;
    label: string;
    flagAssessmentOSModels: FlagAssessmentOSModel[]
}

export interface FlagAssessmentOSModel {
    osName: string | undefined;
    assessment: string;
}
