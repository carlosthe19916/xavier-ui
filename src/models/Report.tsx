export interface Report {
    id: number;
    customerId: string;
    fileName: string;
    numberOfHosts: number;
    totalDiskSpace: number;
    totalPrice: number;
    creationDate: number
}
