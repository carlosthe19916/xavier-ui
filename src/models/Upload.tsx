export interface Upload {
    file: File,
    success: boolean | null,
    error: string | null,
    progress: number,
    uploading: boolean
}
