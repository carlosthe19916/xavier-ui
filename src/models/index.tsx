export * from './Report';
export * from './Upload';
export * from './User';
export * from './Mappings';
