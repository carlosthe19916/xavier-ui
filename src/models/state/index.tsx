import {
    Report,
    ReportInitialSavingEstimation
} from '../Report';
import { User } from '../User';
import { AxiosError } from 'axios';

export type FetchStatus = 'none' | 'inProgress' | 'complete';

export interface CachedReport extends Report {
    timeRequested: number;
}

export interface ReportState {
    byId: Map<string, CachedReport>;
    fetchStatus: Map<string, FetchStatus>;
    errors: Map<string, AxiosError>;
}

export interface ReportListState {
    total: number;
    items: Report[];
    fetchError: AxiosError | null;
    fetchStatus: FetchStatus;
}

export interface CachedReportInitialSavingEstimation extends ReportInitialSavingEstimation {
    timeRequested: number;
}

export interface InitialSavingEstimationState {
    byId: Map<string, CachedReportInitialSavingEstimation>;
    fetchStatus: Map<string, FetchStatus>;
    errors: Map<string, AxiosError>;
}

export interface UploadState {
    uploadFile: File | null;
    uploadStatus: FetchStatus;
    uploadError: AxiosError | null;
    uploadProgress: number;
}

export interface UserState {
    user: User | null;
    fetchError: AxiosError | null;
    fetchStatus: FetchStatus;
}

export type DialogDeleteState = {
    isOpen: boolean;
    isProcessing: boolean;
    isError: boolean;
    name: string;
    type: string;
    onDelete: () => void;
    onCancel: () => void;
};

/**
 * The fields of this interface should match the /store/index.js
 */
export interface GlobalState {
    reportState: ReportState;
    reportListState: ReportListState;
    initialSavingEstimationState: InitialSavingEstimationState;
    uploadState: UploadState;
    userState: UserState;
    dialogDeleteState: DialogDeleteState;
}
