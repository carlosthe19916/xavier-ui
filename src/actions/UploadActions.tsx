import { uploadFile } from '../api/upload';
import { GenericAction } from '../models/action';
import { Upload } from '../models';

export const ActionTypes = {
    UPLOAD_REQUEST: 'UPLOAD_REQUEST',
    UPLOAD_PROGRESS: 'UPLOAD_PROGRESS',
    SELECT_UPLOAD_FILE: 'SELECT_UPLOAD_FILE'
};

export const uploadRequest = (upload: Upload, config = {}): GenericAction => {
    const formData = new FormData();
    formData.append('file', upload.file, upload.file.name);
    formData.append('reportName', upload.reportName);

    if (upload.reportDescription) {
        formData.append('reportDescription', upload.reportDescription);
    }

    if (upload.yearOverYearGrowthRatePercentage) {
        formData.append('reportDesyearOverYearGrowthRatePercentagecription', upload.yearOverYearGrowthRatePercentage.toString());
    }

    if (upload.percentageOfHypervisorsMigratedOnYear1) {
        formData.append('percentageOfHypervisorsMigratedOnYear1', upload.percentageOfHypervisorsMigratedOnYear1.toString());
    }

    if (upload.percentageOfHypervisorsMigratedOnYear2) {
        formData.append('percentageOfHypervisorsMigratedOnYear2', upload.percentageOfHypervisorsMigratedOnYear2.toString());
    }

    if (upload.percentageOfHypervisorsMigratedOnYear1) {
        formData.append('percentageOfHypervisorsMigratedOnYear3', upload.percentageOfHypervisorsMigratedOnYear3.toString());
    }

    return {
        type: ActionTypes.UPLOAD_REQUEST,
        payload: uploadFile(formData, config),
        meta: {
            file: upload.file,
            notifications: {
                rejected: {
                    variant: 'danger',
                    title: `Failed to upload file`
                }
            }
        }
    };
};

export const uploadProgress = (progress: number): GenericAction => ({
    type: ActionTypes.UPLOAD_PROGRESS,
    payload: {
        progress
    }
});

export const selectUploadFile = (file: File): GenericAction => ({
    type: ActionTypes.SELECT_UPLOAD_FILE,
    payload: {
        file
    }
});
