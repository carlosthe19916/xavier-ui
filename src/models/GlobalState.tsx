import { Report } from './Report';
import { Upload } from './Upload';

export interface GlobalState {
    reports: {
        report: Report,
        reports: Report[];
        loading: boolean;
        error: string;
        total: number;
    },
    uploads: {
        uploads: Upload[]
    }
}
