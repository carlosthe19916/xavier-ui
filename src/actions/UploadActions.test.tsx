jest.mock('../api/apiClient', () => ({
    get: jest.fn(() => Promise.resolve({})),
    post: jest.fn(() => Promise.resolve({})),
    put: jest.fn(() => Promise.resolve({})),
    delete: jest.fn(() => Promise.resolve({}))
}));

import * as actionGenerator from './UploadActions';
import { Upload } from '../models';

describe('uploadRequest', () => {
    it('returns a state object', () => {
        const upload: Upload = {
            file: new File([ '' ], 'myFile.zip'),
            name: 'myReport',
            customerId: 'myCustomerId',
            percentageYearOverYearGrowthRate: 5
        };
        expect(actionGenerator.uploadRequest(upload, {})).toMatchSnapshot();
    });
});

describe('uploadProgress', () => {
    it('returns a state object', () => {
        expect(actionGenerator.uploadProgress(new File([ '' ], 'myFile.zip'), 60)).toMatchSnapshot();
    });
});
