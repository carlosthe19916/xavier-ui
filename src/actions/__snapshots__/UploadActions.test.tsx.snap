// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resetUploadFile returns a state object 1`] = `
Object {
  "payload": Object {},
  "type": "RESET_UPLOAD_FILE",
}
`;

exports[`selectFile returns a state object 1`] = `
Object {
  "payload": Object {
    "file": File {},
  },
  "type": "SELECT_UPLOAD_FILE",
}
`;

exports[`updateProgress returns a state object 1`] = `
Object {
  "payload": Object {
    "progress": 60,
  },
  "type": "UPLOAD_PROGRESS",
}
`;

exports[`uploadRequest returns a state object 1`] = `
Object {
  "meta": Object {
    "file": File {},
    "noError": true,
    "notifications": Object {},
  },
  "payload": Promise {},
  "type": "UPLOAD_REQUEST",
}
`;
